### ISSUE TITLE
PyTorch Model Parallel should expose critical metrics to the application.

### ISSUE THREAD

**Issue by pritamdamania87** [pytorch/pytorch#28245]
## 🚀 Feature

Context for Model Parallel: https://github.com/pytorch/pytorch/issues/23110

## Motivation

When applications are using complex distributed primitives like RPC, RRef and Distributed Autograd, debugging issues can be cumbersome. We should have a way of exposing metrics to applications. This could simply be a `def get_metrics() -> Dict[str, int]` API that returns information about various things. The full list of metrics needs to be decided, although a few examples could be number of owner rrefs, number of user rrefs, RPC latency, Distributed autograd latency etc.

**Comment by gchanan** [comment]
@smessmer I believe you are supposed to put at least one module on an issue before marking it as triaged.

**Comment by ezyang** [comment]
So... is model parallel the same team as distributed? If not, we should promote model-parallel to a top level label, same as distributed.

**Comment by mrshenli** [comment]
Updated `topic: model-parallel` to `module: rpc`

**Comment by rohan-varma** [comment]
If the high-priority label means that it should be shipped by the next release of pytorch, then we should remove the label, since this will likely not be ready in time for the release

**Comment by ezyang** [comment]
@rohan-varma no, it doesn't necessarily mean that. It's more of a way to make sure we don't forget about these bugs; we do metrics on high priority issues, we don't on non-high priority issues.
